File: 시저암호/java/20221021/src/SolutionTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    @Test
    void moveCharacter() {
        Solution solution = new Solution();

        assertEquals("B", solution.moveCharacter("A", 1));
        assertEquals("Z", solution.moveCharacter("A", 25));
        assertEquals("z", solution.moveCharacter("a", 25));
        assertEquals("y", solution.moveCharacter("z", 25));
        assertEquals("x", solution.moveCharacter("y", 25));
    }

    @Test
    void convert() {
        Solution solution = new Solution();

        assertEquals("BC", solution.convert("AB", 1));
        assertEquals("a", solution.convert("z", 1));
        assertEquals("e F d", solution.convert("a B z", 4));
    }
}

File: 시저암호/javascript/20221021/src/solution.test.js
import convert, { moveCharacter } from './solution';

test('moveCharacter', () => {
  expect(moveCharacter({ character: 'A', times: 1 })).toBe('B');
  expect(moveCharacter({ character: 'Z', times: 1 })).toBe('A');
  expect(moveCharacter({ character: 'z', times: 1 })).toBe('a');
  expect(moveCharacter({ character: 'y', times: 2 })).toBe('a');
  expect(moveCharacter({ character: 'a', times: 2 })).toBe('c');
  expect(moveCharacter({ character: 'a', times: 25 })).toBe('z');
});

test('convert', () => {
  expect(convert({ string: 'AB', times: 1 })).toBe('BC');
  expect(convert({ string: 'z', times: 1 })).toBe('a');
  expect(convert({ string: 'a B z', times: 4 })).toBe('e F d');
  expect(convert({ string: ' a B z ', times: 4 })).toBe(' e F d ');
  expect(convert({ string: ' a B z ', times: 25 })).toBe(' z A y ');
});
